applications/Chat/coati/dataset/sft_dataset.py

```diff
@@ -127,15 +180,27 @@ def __init__(self, data_path: str, tokenizer: transformers.PreTrainedTokenizer,
             list_data_dict = list_data_dict[:max_datasets_size]
 
         logger.info("Formatting inputs...")
-        prompt_input, prompt_no_input = PROMPT_DICT["prompt_input"], PROMPT_DICT["prompt_no_input"]
-        sources = [
-            prompt_input.format_map(example) if example.get("input", "") != "" else prompt_no_input.format_map(example)
-            for example in list_data_dict
-        ]
-        targets = [f"{example['output']}{tokenizer.eos_token}" for example in list_data_dict]
-
-        logger.info("Tokenizing inputs... This may take some time...")
-        data_dict = preprocess(sources, targets, tokenizer, max_length)
+        if "conversations" not in list_data_dict[0]:
+            prompt_input, prompt_no_input = PROMPT_DICT["prompt_input"], PROMPT_DICT["prompt_no_input"]
+            sources = [
+                prompt_input.format_map(example)
+                if example.get("input", "") != "" else prompt_no_input.format_map(example) for example in list_data_dict
+            ]
+            targets = [f"{example['output']}{tokenizer.eos_token}" for example in list_data_dict]
+
+            if is_rank_0():
+                logger.info("Tokenizing inputs... This may take some time...")
+
+            data_dict = preprocess(sources, targets, tokenizer, max_length)
+        else:
+            if is_rank_0():
+                logger.info("Tokenizing inputs... This may take some time...")
+
+            sources = [conv["conversations"] for conv in list_data_dict]
+            data_dict = preprocess_conversation(sources, tokenizer, max_length)
```

Comment:
How can we trigger conversation-style dataset? 
Is it possible to provide an example in the README file?